pkg/networkserver/grpc_gsns.go

```diff
@@ -1135,9 +1136,13 @@ func (ns *NetworkServer) handleJoinRequest(ctx context.Context, up *ttnpb.Uplink
 		RawPayload:         up.RawPayload,
 		RxDelay:            macState.DesiredParameters.Rx1Delay,
 		SelectedMacVersion: matched.LorawanVersion, // Assume NS version is always higher than the version of the device
-		DownlinkSettings:   dlSettings,
-		ConsumedAirtime:    up.ConsumedAirtime,
-	})
+		DownlinkSettings:   &dlSettings,
+	}
+	if up.ConsumedAirtime != nil {
+		jReq.ConsumedAirtime = ttnpb.ProtoDuration(up.ConsumedAirtime)
```

Comment:
Same here - `ProtoDurationOrZero` does this (0 if `nil`).

Reply:
`ProtoDurationOrZero`: There's no such function.

Reply:
I was wrong - your original code was correct - there is no need for this function.